website/docs/concepts/event-bridge/event-bus.md

```diff
@@ -6,343 +6,125 @@ sidebar_position: 1
-    event.detail.age >= 18 &&
-    event.detail.age < 30
-);
-```
-
-Rules can be further refined by calling `when` on a Functionless `EventBusRule`.
-
-```ts
-// Cat people who are between 18 and 30 and do not also like dogs.
-catPeopleEvents.when((event) => !event.detail.interests.includes("DOGS"));
-```
-
-## Transform the event before sending to some services like `Lambda` Functions.
-
-We have two lambda functions to invoke, one for create or updates and another for deletes, lets make those.
-
-```ts
-const createOrUpdateFunction = new aws_lambda.Function(this, 'createOrUpdate', ...);
-const deleteFunction = new aws_lambda.Function(this, 'delete', ...);
-```
-
-and wrap them with Functionless's `Function` wrapper, including given them input types.
-
-```ts
-interface CreateOrUpdate {
-  id?: string;
-  name: string;
-  age: number;
-  operation: "Create" | "Update";
-  interests: string[];
-}
-
-interface Delete {
-  id: string;
-}
-
-const createOrUpdateOperation = functionless.Function<CreateOrUpdate, void>(
-  createOrUpdateFunction
-);
-const deleteOperation = functionless.Function<Delete, void>(deleteFunction);
-```
-
-The events from before do not match the formats from before, so lets transform them to the structures match.
-
-```ts
-const createOrUpdateEventsTransformed =
-  createOrUpdateEvents.map<CreateOrUpdate>((event) => ({
-    id: event.detail.id,
-    name: event.detail.name,
-    age: event.detail.age,
-    operation: event["detail-type"],
-    interests: event.detail.interests,
-  }));
-
-const deleteEventsTransformed = createOrUpdateEvents.map<Delete>((event) => ({
-  id: event.detail.id,
-}));
-```
-
-## Target other AWS services like Lambda and other Event Buses
-
-Now that we have created rules on our event buses using `when` and transformed those matched events using `map`, we need to send the events somewhere.
+- Step Functions
+- Lambda
+- Event Bus
 
-We can `pipe` the transformed events to the lambda functions we defined earlier.
+### Step Functions
 
 ```ts
-createOrUpdateEventsTransformed.pipe(createOrUpdateOperation);
-deleteEventsTransformed.pipe(deleteOperation);
+const bus = new EventBus();
+const sfn = new StepFunction(stack, "sfn", () => {
+  bus({
+    source: "myStepFunction",
+    "detail-type": "someType",
+    detail: {},
+  });
+});
 ```
 
-What about our young cat lovers? We want to forward those events to our sister team's event bus for processing.
+:::caution
+Caveat: Events passed to the bus in a step function must be one or more literal objects and may not use the spread (`...`) syntax.
 
 ```ts
-const catPeopleBus = functionless.EventBus.fromBus(
-  aws_events.EventBus.fromEventBusArn(this, "catTeamBus", catTeamBusArn)
-);
-
-// Note: EventBridge does not support transforming events which target other event buses. These events are sent as is.
-catPeopleEvents.pipe(catPeopleBus);
+const event = { source: "lambda", "detail-type": "type", detail: {} };
+bus(event); // error
```

Comment:
Maybe it should be `bus.emit`?

Reply:
1. What is the logic for when direct invocation can be utilized vs a method?
2. Why use `emit` vs `putEvents`?